### ISSUE TITLE
Full Text Search on local RTD instance not working

### ISSUE THREAD

**Issue by frodopwns** [rtfd/readthedocs.org#1638]
When I spin up RTD locally on a Ubuntu VM and turn on ElasticSearch I can search titles of projects I have imported but when I go to a project page /projects/projecT_name/search I cannot find anything within the search results.

When I examine ElasticSearch I see items in the readthedocs index of type project but no entries of the type page that I would expect need to be there for full page search.

I found some instructions here: https://github.com/rtfd/readthedocs.org/blob/master/docs/rtfd.rst

But when I run `index.create_index(index_name)` I get the error:
TransportError: TransportError(400, u'IndexCreationException[[readthedocs-20150909120501] failed to create index]; nested: IllegalArgumentException[Custom Analyzer [default_icu] failed to find tokenizer under name [icu_tokenizer]]; ')

Any help in getting this functionality would be greatly appreciated.

**Comment by agjohnson** [comment]
Supporting this on local instances is outside the scope of the core team, marking this community support if anyone else wants to jump in on this.

**Comment by frodopwns** [comment]
So does that mean it either should work out of the box or is too complicated to explain?

**Comment by agjohnson** [comment]
It should Just Work, as it does in our environments, but it's also too complicated for the core team to spend time supporting these external instances that we have no control over. If you are running an internal instance, you should be familiar with managing the entire stack outside of Read the Docs.

We answered this question here:
http://docs.readthedocs.org/en/latest/open-source-philosophy.html

This isn't to say we won't be able to help, but it's effectively very low on a long list and better served by the community.

**Comment by frodopwns** [comment]
Being familiar with managing the stack I get the feeling this is a documentation issue which is ironic given your area of business.

**Comment by frodopwns** [comment]
It would make a lot of sense for your project to get this thing running in a VM via ansible or Chef...then you dont have to help users run the project anymore...might increase your pull requests.

**Comment by domdinicola** [comment]
@frodopwns I've experimented the same problem... did you find a solution?

I've compiled elasticsearch-1.7.0 and elasticsearch-analysis-icu-2.3.0

in my virtualenv as for readthedocs requirements
elasticsearch==1.5.0
pyelasticsearch==0.7.1
django-haystack==2.1.0
celery-haystack==0.7.2

I also tried upgrading libraries 
elasticsearch==1.6.0
pyelasticsearch==1.4
celery-haystack==0.9
django-haystack==2.1.1

but isn't helping...

Any suggestion would be appreciated!

**Comment by frodopwns** [comment]
@domdinicola No I decided to throw together my own solution instead of using RTD. They must have a very manual setup for ES that isn't well documented.

**Comment by domdinicola** [comment]
I did the same... overridden view and template...
Thank you for answering.

**Comment by agjohnson** [comment]
Our deployment of elastic search is documented in our production docs, if anyone else is searching. The set up only requires ES + icu, and index creation.

Closing this due to inactivity.

**Comment by acjackson** [comment]
Where can I find these production docs?

**Comment by RichCastar** [comment]
@acjackson, by production docs, I think @agjohnson means this: http://read-the-docs.readthedocs.io/en/latest/rtfd.html#elastic-search-setup.  Those are the only docs on elasticsearch setup, AFAICT.

I had zero problems with the rest of the install, and I  have a local instance of RTD running and working (even git pulling, building, and running doxygen and breathe on stuff). Installing elasticsearch was a snap as well, but getting that code to run and do anything meaningful to the elasticsearch index is a different story (that covers many late nights, by many a sad, sad tech writer).

**Comment by atcase** [comment]
I got this working using elasticsearch 1.7.6 and icu plugin 2.7.0. I struggled for a while trying to use the latest elasticsearch, but it seems somewhat incompatible, and based on the readthedocs.org documentation suggesting the icu plugin 2.30 I suspect it is using an even older elasticsearch.

The elasticsearch/icu compatibilty table is at https://github.com/elastic/elasticsearch-analysis-icu

The docs on readthedocs.org elastic search are at http://readthedocsorg.readthedocs.io/en/latest/rtfd.html?highlight=elastic#elastic-search

Anyway, the steps I used.

1. Install elasticsearch 1.7.6
2. Install icu plugin 2.7.0 using`/usr/share/elasticsearch/bin/plugin -install elasticsearch/elasticsearch-analysis-icu/2.7.0`
3. fire up elasticsearch
4. Run magic python script from rtfd page above, slightly modified:
```
import os
os.environ['DJANGO_SETTINGS_MODULE'] = 'readthedocs.settings.dev'
from readthedocs.search.indexes import Index, PageIndex, ProjectIndex, SectionIndex
index = Index()
index_name = index.timestamped_index()
index.create_index(index_name)
index.update_aliases(index_name)
proj = ProjectIndex()
proj.put_mapping()
page = PageIndex()
page.put_mapping()
sec = SectionIndex()
sec.put_mapping()
```
5. Run `python manage.py reindex_elasticsearch`
